pkg/ccl/backupccl/restore_data_processor.go

```diff
@@ -343,10 +460,13 @@ func (rd *restoreDataProcessor) openSSTs(
 	if err != nil {
 		return err
 	}
-	return sendIter(iter, dirs)
+
+	return sendIter(iter, dirs, sstCh, iterAllocs, true)
 }
 
-func (rd *restoreDataProcessor) runRestoreWorkers(ctx context.Context, ssts chan mergedSST) error {
+func (rd *restoreDataProcessor) runRestoreWorkers(
+	ctx context.Context, ssts chan mergedSST, workerSSTChannels []chan mergedSST,
+) error {
 	return ctxgroup.GroupWorkers(ctx, rd.numWorkers, func(ctx context.Context, worker int) error {
 		kr, err := MakeKeyRewriterFromRekeys(rd.FlowCtx.Codec(), rd.spec.TableRekeys, rd.spec.TenantRekeys,
```

Comment:
maybe here we should do ```if err := mem.Grow(minWorkerMemReservation); err != nil { if worker != 0 { return nil // no memory to run this worker} return errors.Wrap(err, "insufficient memory available to run restore") } else { // add minWorkerMemReservation to the pool``` to ensure the pool has a baseline amount (say 25MB) per worker ?

Reply:
done, reserved 25MB at the start of the worker.